### PR TITLE
Don't mention the nonexistant 'lifetimes guide'

### LINKED ISSUE #22003: Documentation (The Book): Macros chapter mentions non-existing (?) "lifetime guide"
The chapter on macros <http://doc.rust-lang.org/book/macros.html#hygiene> mentions a "lifetime guide" when using loop labels. As far as I can tell, that guide does not exist anymore (or it has been renamed when becoming a chapter). If that's correct, then loop labels are not mentioned anywhere else in the book, so this subsection is rather hard to follow. If the lifetime guide still exists somewhere, but has been renamed, then the text should be updated (preferably adding a link).

### PR DESCRIPTION
**Pull request by steveklabnik**
Fixes #22003

### CODE REVIEW

**Comment by Manishearth** [comment]
Needs a rebase, sorry.

**Comment by steveklabnik** [comment]
This has effectively been fixed by other work.
